### PR TITLE
Review the `bp_view` capability support in endpoints

### LINKED ISSUE #496: Review the `bp_view` capability support in endpoints
After adding support for the GraphQL API, https://github.com/renatonascalves/wp-graphql-buddypress/pull/124, I'm pretty sure we are missing support in a few endpoints.

See #487

### PR DESCRIPTION
**Pull request by renatonascalves**
- fixes #496 
- adds unit tests
- add support for missing components, i.e., activity

### CODE REVIEW

**Comment by renatonascalves** [comment]
Just a note that the GraphQL API, https://github.com/renatonascalves/wp-graphql-buddypress/pull/124, took a different approach. We don't return an error but rather an empty result. Mainly because in GraphQL, fields can be nested ([connections](https://www.apollographql.com/blog/explaining-graphql-connections)).

File: includes/bp-members/classes/class-bp-rest-signup-endpoint.php [hunk 1]
```diff
@@ -203,20 +203,18 @@ public function get_items_permissions_check( $request ) {
 			array( 'status' => rest_authorization_required_code() )
 		);
 
-		$capability = 'edit_users';
+		if ( bp_current_user_can( 'bp_view', array( 'bp_component' => 'members' ) ) ) {
```
Comment (imath) [inline includes/bp-members/classes/class-bp-rest-signup-endpoint.php:206]: I think this part is not necessary as we need a higher cap to get signups, the one you are using at line 207.

File: includes/bp-members/classes/class-bp-rest-signup-endpoint.php [hunk 2]
```diff
@@ -272,19 +270,35 @@ public function get_item( $request ) {
...
-		$retval = $this->get_items_permissions_check( $request );
+		$retval = new WP_Error(
+			'bp_rest_authorization_required',
+			__( 'Sorry, you are not authorized to perform this action.', 'buddypress' ),
+			array( 'status' => rest_authorization_required_code() )
+		);
 
-		if ( ! is_wp_error( $retval ) ) {
-			$signup = $this->get_signup_object( $request->get_param( 'id' ) );
+		if ( bp_current_user_can( 'bp_view', array( 'bp_component' => 'members' ) ) ) {
```
Comment (imath) [inline includes/bp-members/classes/class-bp-rest-signup-endpoint.php:279]: I think this part is not necessary as we need a higher cap to get signups, the one you are using at line 280.

**Review by imath — COMMENTED** [review]
Hi @renatonascalves another great job on this one 💪 . I would only remove the `'bp_view'` check cap you added into the signups endpoint as anyway a higher cap is needed (`'edit_users'` or `'manage_network_users'`)

File: includes/bp-members/classes/class-bp-rest-signup-endpoint.php (hunk 1 above)
Comment (renatonascalves) [reply includes/bp-members/classes/class-bp-rest-signup-endpoint.php]: Good call.

**Pull request merged**